src/Cache/Adapter/Sharding.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Utopia\Cache\Adapter;
+
+use Utopia\Cache\Adapter;
+
+class Sharding implements Adapter
+{
+    /**
+     * @var Adapter[] 
+     */
+    protected $adapters;
+
+    /**
+     * Sharding constructor.
+     * 
+     * Allows to shard cache across multiple adapters in a consistent way.
+     * 
+     * Please Note:
+     *  When adding a new adapter to the pool, cached will gradually
+     *  get re-distributed to fill the new adapter, this might casue a siegnificant drop
+     *  in hit-rate for a short period of time.
+     * 
+     * @param Adapter[] $adapters
+     */
+    public function __construct(array $adapters)
+    {
+        $this->adapters = $adapters;
+    }
+
+    /**
+     * @param string $key
+     * @param int $ttl time in seconds
+     * @return mixed
+     */
+    public function load($key, $ttl)
+    {
+        return $this->getAdapter($key)->load($key, $ttl);
+    }
+
+    /**
+     * @param string $key
+     * @param string|array $data
+     * @return bool|string|array
+     */
+    public function save($key, $data)
```

Comment:
```suggestion
    public function save(string $key, string|mixed $data): bool|string|array
```
looking at this method types, I don't think it looks good. We should think about if we can improve this in the future.
1. taking a string or an array
2. returning bool,string or array

There is so much going on, that using it has many open questions.